src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1964,17 +1963,21 @@ private void CheckRecordMemberNames(BindingDiagnosticBag diagnostics)
             }
         }
 
-        private void CheckMemberNameConflicts(BindingDiagnosticBag diagnostics)
+        private static void CheckMemberNameConflicts(
+            SourceMemberContainerTypeSymbol containerForDiagnostics,
```

Comment:
nit: I didn't follow why this method was changed to static. The only value passed to `containerForDiagnostics` is `this`. 
The same applies to `CheckIndexerNameConflicts` and `ReportMethodSignatureCollision`

Reply:
> nit: I didn't follow why this method was changed to static. The only value passed to `containerForDiagnostics` is `this`. The same applies to `CheckIndexerNameConflicts` and `ReportMethodSignatureCollision`

There are additional parameters passed as well. While at the moment the only value passed to `containerForDiagnostics` is `this`, it doesn't have to be `this`, the logic doesn't depend on the fact. The goal was to make sure that the methods do not accidentally use any instance API any instance data, they should operate only on the information passed in explicitly and shouldn't assume that the passed in symbols directly declared in `this`.